TheMovieDb/Home/Contoller/HomeViewController.swift

```diff
@@ -7,21 +7,222 @@
+    private func fetchUpcoming() {
+        group.enter()
+        MovieAPI.shared.getUpcoming(completion: {(response: Result<Movies, Error>) in
+            switch response {
+            case .failure(let error):
+                debugPrint(error)
+            case .success(let res):
+                self.upComingMovies =  res.movies
+            }
+            self.group.leave()
+        })
+    }
+    
+    private func fetchTrending() {
+        group.enter()
+         MovieAPI.shared.getTrending(completion: {(response: Result<Movies, Error>) in
+             switch response {
+             case .failure(let error):
+                 debugPrint(error)
+             case .success(let res):
+                 self.trendingMovies = res.movies
+             }
+             self.group.leave()
+         })
+    }
+
     // MARK: - Helpers
+    private func configureUICollection() {
+        collectionView.register(HightSectionCell.self, forCellWithReuseIdentifier: HightSectionCell.reuseIdentifier)
+        collectionView.register(DefaultSectionCell.self, forCellWithReuseIdentifier: DefaultSectionCell.reuseIdentifier)
+        collectionView.register(TopRatedSectionCell.self, forCellWithReuseIdentifier: TopRatedSectionCell.reuseIdentifier)
+        collectionView.register(HomeHeader.self, forSupplementaryViewOfKind: categoryHomeHeaderId, withReuseIdentifier: HomeHeader.reuseIdentifier)
+    }
+    
+    private func getCurrentMovie(indexPath: IndexPath) -> Movie {
+        let section = MovieGroupSections(rawValue: indexPath.section) ?? .trending
+        var movie: Movie
+        switch section {
+        case .popular:
+            movie = popularMovies[indexPath.row]
+        case .trending:
+            movie = trendingMovies[indexPath.row]
+        case .playingNow:
+            movie = playingNowMovies[indexPath.row]
+        case .topRated:
+            movie = topRatedMovies[indexPath.row]
+        case .upcoming:
+            movie = upComingMovies[indexPath.row]
+        }
+        return movie
+    }
+    
     private func configureUI() {
-        view.backgroundColor = .systemBackground
+        navigationItem.title = "Movies"
+        view.backgroundColor = .black
     }
-    // MARK: - Actions
+    
+}
 
+// MARK: - UIControllerViewDataSource
+extension HomeViewController {
+    override func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        let section = MovieGroupSections(rawValue: section) ?? .trending
+        switch section {
+        case .popular:
+            return popularMovies.count
+        case .trending:
+            return trendingMovies.count
+        case .playingNow:
+            return playingNowMovies.count
+        case .topRated:
+            return topRatedMovies.count
+        case .upcoming:
+            return upComingMovies.count
+        }
+    }
+    
+    override func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return 5
```

Comment:
If we have a MovieGroupSections. Can we use it to take from there the number of sections?